rust-runtime/aws-smithy-types/src/primitive.rs

```diff
@@ -107,25 +119,25 @@ pub enum Encoder {
     Bool(bool),
     /// 8-bit signed integer
     #[non_exhaustive]
-    I8(i8, itoa::Buffer),
```

Comment:
do we actually even need these public variants on encoder? Can we do `struct Encoder(Inner); `enum Inner { ... }`?

Reply:
since we have the from impls already

Reply:
I like this idea! Changed over.